tools/traffic/workers/blob_status_tracker.go

```diff
@@ -0,0 +1,253 @@
+		getStatusErrorCountMetric:         generatorMetrics.NewCountMetric("get_status_ERROR"),
+		unknownCountMetric:                generatorMetrics.NewCountMetric("get_status_UNKNOWN"),
+		processingCountMetric:             generatorMetrics.NewCountMetric("get_status_PROCESSING"),
+		dispersingCountMetric:             generatorMetrics.NewCountMetric("get_status_DISPERSING"),
+		failedCountMetric:                 generatorMetrics.NewCountMetric("get_status_FAILED"),
+		insufficientSignaturesCountMetric: generatorMetrics.NewCountMetric("get_status_INSUFFICIENT_SIGNATURES"),
+		confirmedCountMetric:              generatorMetrics.NewCountMetric("get_status_CONFIRMED"),
+		finalizedCountMetric:              generatorMetrics.NewCountMetric("get_status_FINALIZED"),
+	}
+}
+
+// Start begins the status goroutine, which periodically polls
+// the disperser service to verify the status of blobs.
+func (verifier *BlobStatusTracker) Start() {
+	verifier.waitGroup.Add(1)
+	go verifier.monitor()
+}
+
+// monitor periodically polls the disperser service to verify the status of blobs.
+func (verifier *BlobStatusTracker) monitor() {
+	ticker := time.NewTicker(verifier.config.VerifierInterval)
+	for {
+		select {
+		case <-(*verifier.ctx).Done():
+			verifier.waitGroup.Done()
+			return
+		case key := <-verifier.keyChannel:
+			verifier.unconfirmedBlobs = append(verifier.unconfirmedBlobs, key)
+		case <-ticker.C:
+			verifier.poll()
+		}
+	}
+}
+
+// poll checks all unconfirmed keys to see if they have been confirmed by the disperser service.
+// If a Key is confirmed, it is added to the blob confirmedBlobs and removed from the list of unconfirmed keys.
+func (verifier *BlobStatusTracker) poll() {
+
+	// FUTURE WORK If the number of unconfirmed blobs is high and the time to confirm is high, this is not efficient.
+	// Revisit this method if there are performance problems.
+
+	nonFinalBlobs := make([]*UnconfirmedKey, 0)
+	for _, key := range verifier.unconfirmedBlobs {
+
+		blobStatus := verifier.getBlobStatus(key)
+		if blobStatus == nil {
+			// There was an error getting status. Try again later.
+			nonFinalBlobs = append(nonFinalBlobs, key)
+			continue
+		}
+
+		verifier.updateStatusMetrics(blobStatus.Status)
+		if isBlobStatusTerminal(blobStatus.Status) {
+			if isBlobStatusConfirmed(blobStatus.Status) {
+				verifier.forwardToReader(key, blobStatus)
+			}
+		} else {
+			// try again later
+			nonFinalBlobs = append(nonFinalBlobs, key)
+		}
+	}
+	verifier.unconfirmedBlobs = nonFinalBlobs
+	verifier.blobsInFlightMetric.Set(float64(len(verifier.unconfirmedBlobs)))
+}
+
+// isBlobStatusTerminal returns true if the status is a terminal status.
+func isBlobStatusTerminal(status disperser.BlobStatus) bool {
+	switch status {
+	case disperser.BlobStatus_FAILED:
+		return true
+	case disperser.BlobStatus_INSUFFICIENT_SIGNATURES:
+		return true
+	case disperser.BlobStatus_CONFIRMED:
```

Comment:
CONFIRMED isn't a terminal state as it can be transitioned to FINALIZED or FAILED (if reorg'd)

Reply:
Correct, although the behavior is intentional in this context. Added a comment to clarify.

```
case disperser.BlobStatus_CONFIRMED:
		// Technically this isn't terminal, as confirmed blobs eventually should become finalized.
		// But it is terminal from the status tracker's perspective, since we stop tracking the blob
		// once it becomes either confirmed or finalized.
		return true
```